apps/torus-allocator/src/app/(expanded-pages)/agent/[slug]/components/update-agent-button/update-agent-dialog/update-agent-dialog.tsx

```diff
@@ -161,7 +166,12 @@ export default function UpdateAgentDialog({
         form={form}
         updateAgentMutation={updateAgentMutation}
         imageFile={imageFile}
-        hasUnsavedChanges={form.formState.isDirty}
+        hasUnsavedChanges={
+          form.formState.isDirty &&
+          Object.keys(form.formState.dirtyFields).some(
+            (field) => field !== "name",
+          )
```

Comment:
If this filter is not redundant, abstract it into a function as it appears too in L96.